ember-core/shared/src/main/scala/org/http4s/ember/core/Encoder.scala

```diff
@@ -58,8 +59,12 @@ private[ember] object Encoder {
         ()
       }
     }
-    if (!chunked && !appliedContentLength && resp.status.isEntityAllowed) {
-      stringBuilder.append(chunkedTansferEncodingHeaderRaw).append(CRLF)
+    if (!appliedContentLength && resp.entity == Entity.Empty && resp.status.isEntityAllowed) {
+      stringBuilder.append(zeroContentLengthRaw).append(CRLF)
+      chunked = false
+      ()
+    } else if (!chunked && !appliedContentLength && resp.status.isEntityAllowed) {
+      stringBuilder.append(chunkedTransferEncodingHeaderRaw).append(CRLF)
       chunked = true
       ()
     }
```

Comment:
Note the follow-up work from https://github.com/http4s/http4s/pull/6444#issuecomment-1146660863.